src/ExpressionEvaluator/Core/Source/ResultProvider/Formatter.TypeNames.cs

```diff
@@ -25,10 +24,36 @@ internal string GetTypeName(TypeAndCustomInfo typeAndInfo, bool escapeKeywordIde
             var dynamicFlags = DynamicFlagsCustomTypeInfo.Create(typeAndInfo.Info);
             var index = 0;
             var pooled = PooledStringBuilder.GetInstance();
-            AppendQualifiedTypeName(pooled.Builder, type, dynamicFlags, ref index, escapeKeywordIdentifiers);
+            AppendQualifiedTypeName(pooled.Builder, type, dynamicFlags, ref index, escapeKeywordIdentifiers, out sawInvalidIdentifier);
             return pooled.ToStringAndFree();
         }
 
+        /// <remarks>
+        /// Stops at the first backtick.
+        /// </remarks>
+        internal bool IsValidIdentifier(string name)
+        {
+            if (!IsIdentifierStartCharacter(name[0]))
+            {
+                return false;
+            }
+
+            foreach (char ch in name)
+            {
+                if (ch == '`')
+                {
+                    break;
```

Comment:
I've moved the trimming into the caller.  It happens for generic types, but we'll flag member names containing backticks.